sql/src/test/java/io/crate/operation/projectors/FetchProjectorTest.java

```diff
@@ -76,32 +76,35 @@ public void after() throws Exception {
     }
 
     @Test
-    public void testFetchSize() throws Throwable {
+    public void testMultipleFetchRequests() throws Throwable {
         CollectingRowReceiver rowReceiver = new CollectingRowReceiver();
 
-        FetchOperation fetchOperation = mock(FetchOperation.class);
-        SettableFuture<IntObjectMap<? extends Bucket>> future = SettableFuture.create();
-        when(fetchOperation.fetch(anyString(), any(IntObjectMap.class), anyBoolean()))
-            .thenReturn(future);
+        // dummy FetchOperation that returns buckets for each reader-id where each row contains a column that is the same as the docId
+        FetchOperation fetchOperation = new FetchOperation() {
+            @Override
+            public ListenableFuture<IntObjectMap<? extends Bucket>> fetch(String nodeId, IntObjectMap<? extends IntContainer> toFetch, boolean closeContext) {
+                IntObjectHashMap<Bucket> readerToBuckets = new IntObjectHashMap<>();
+                for (IntObjectCursor<? extends IntContainer> cursor : toFetch) {
+                    List<Object[]> rows = new ArrayList<>();
+                    for (IntCursor docIdCursor : cursor.value) {
+                        rows.add(new Object[] { docIdCursor.value });
+                    }
+                    readerToBuckets.put(cursor.key, new CollectionBucket(rows));
+                }
+                return Futures.<IntObjectMap<? extends Bucket>>immediateFuture(readerToBuckets);
+            }
+        };
         RowUpstream upstream = mock(RowUpstream.class);
 
         int fetchSize = 3;
         FetchProjector fetchProjector = prepareFetchProjector(fetchSize, rowReceiver, fetchOperation, upstream);
-
-        IntObjectMap<Bucket> intObjectMap = new IntObjectHashMap<>(2);
-        Collection<Object[]> rows = new ArrayList<>(10);
-
         long i;
         for (i = 1; i <= 10; i++) {
             Row row = new Row1(i);
-            rows.add(row.materialize());
-            CollectionBucket collectionBucket = new CollectionBucket(rows);
-            intObjectMap.put(0 , collectionBucket);
-
             fetchProjector.setNextRow(row);
+
             if (i % fetchSize == 0) {
                 verify(upstream, timeout(1000).times((int)i / fetchSize)).pause();
-                future.set(intObjectMap);
                 verify(upstream, timeout(1000).times((int)i / fetchSize)).resume(false);
```

Comment:
Do you think it makes sense to reduce the timeout now with the dummy fetchOperation?